### ISSUE TITLE
New user registration UI

### ISSUE THREAD

**Issue by nbtdev** [nbtdev/client#2]
This does not need to be a component; it can be a separate page. It should use AngularJS for functionality. This page will be presented when a user clicks the "Register" link on the login form in https://github.com/nbtdev/client/issues/1. This link corresponds to a "register" hypermedia type, for which there will be a link from the login form.

Typical new-user registration fields/controls, including:
- account (login) name
- email address
- password
- password verify
- Submit button (submit registration data to API)
- Cancel button (simple "Back" browser behavior)

Underneath this form should be a link labeled "Privacy Policy", which will, when clicked, open a popup window that displays the contents found through the "privacy" hypermedia link.

The form should also include a captcha (Google reCAPTCHA preferred) underneath the registration form, I will provide the unique hash assigned to netbattletech.com (you may need to register your own, or test with the captcha disabled). If the user has not completed the captcha, then Submit functionality is disabled. 

The page should verify that the account name only contains alphanumeric (UTF-8) characters, plus dash (-) and underscore (_). Spaces in account logins are not allowed.

The page should check the account name to make sure it is not already used. This should be done via REST call to NBT API when the user moves focus off of the login field. The user should see feedback (either textual or graphic, i.e. green checkmark or red X), depending on whether the login is already used). 

The page should check to make sure that the contents of the password fields match; similar feedback as described above should be provided if (a) there is text in both fields, when (b) the user moves focus out of one of them. 

The page should check to make sure the email address is valid. This should be a syntax check to make sure it's a valid form (name@domain.ext, etc.). Feedback should be provided, similar to that described above.

When all fields are valid and the captcha is completed, the user will be able to press the Submit button, which forwards the information to the backend API. The user is then informed that an additional step is required, and they should check their email inbox for an account activation link.

**Comment by nbtdev** [comment]
This is largely implemented. Remaining steps are 
- "forgot password" workflow
- "forgot username" workflow

The first one, I expect, will require a login (username, not callsign) and email address, and will send the user a link similar to what they get for new-account reg, but will allow them to change their password.

The second will require just an email address, and will only email the current login (not callsign) to the user. 

Both of these will have captchas on the page (similar to the new user reg) to prevent random bots from spamming the automation API with bogus reset requests.

A user's password will NOT be changed by putting the account into "recovery" mode; if a user manages to log in after sending a password recovery request, we will put the account back into "active" mode.

**Comment by nbtdev** [comment]
Depends on #5

**Comment by nbtdev** [comment]
#5 is in, and closing this even though the last comment is not addressed (moving that to a separate task https://github.com/nbtdev/client/issues/6)
